tests/testapp/test_models.py

```diff
@@ -119,6 +120,13 @@ def test_approx_count_with_straight_join(self):
 
 class QueryHintTests(TestCase):
 
+    def _skip_for_mysql8(self):
+        if (
+            not connection_is_mariadb(connection) and
+            connection.mysql_version >= (8, 0)
+        ):
+            raise SkipTest("Query cache removed in MySQL 8.0")
```

Comment:
Nice, but a decorator applied to the tests would be a more conventional way of doing this, imitating `skipUnless`, which we can't use directly as it takes a condition evaluated at import time and we can't connect to DB until runtime